lightningd/payalgo.c

```diff
@@ -514,13 +516,29 @@ static bool json_pay_try(struct pay *pay)
 	/* Generate random seed */
 	randombytes_buf(&seed, sizeof(seed));
 
+	/* Generate an overpayment, from fuzz * maxfee. */
+	/* Now normally the use of double for money is very bad.
+	 * Note however that a later stage will ensure that
+	 * we do not end up paying more than maxfeepercent
+	 * of the msatoshi we intend to pay. */
+	maxoverpayment = ((double) pay->msatoshi * pay->fuzz * pay->maxfeepercent)
+		/ 100.0;
+	if (maxoverpayment > 0) {
+		randombytes_buf(&overpayment, sizeof(overpayment));
+		/* We will never generate the maximum computed
+		 * overpayment this way. Maybe OK for most
+		 * purposes. */
+		overpayment = overpayment % maxoverpayment;
```

Comment:
We have pseudorand for this, which makes this code neater and also indicates we don't need cryptographic randomness.  We only use randombytes_buf() above because we don't have a routine to give us a pseudorandom siphash seed, but that would be trivial to implement too.

```
if (maxoverpayment > 0)
     overpayment = pseudorand(maxoverpayment);
else
    overpayment = 0;
```